### ISSUE TITLE
Expandable code blocks

### ISSUE THREAD

**Issue by hellt** [squidfunk/mkdocs-material#2171]
- [x] I've read the [contribution guidelines][1] and agree with them

__I want to suggest an idea and checked that ...__

- [x] ... to my best knowledge, my idea wouldn't break something for other users
- [x] ... the documentation does not mention anything about my idea
- [x] ... there are no open or closed issues that are related to my idea

## Description

Implement an expandable code block as demonstrate here

https://user-images.githubusercontent.com/5679861/103466090-ee3b2a80-4d41-11eb-98b7-34b0b4702d23.mp4

### Use Cases

The width of the main div is ok for text, but quite often the code and/or configuration snippets are wider than the visible area.
To make these code blocks less painful to read the proposal is to make the code blocks expandable on user action.

**Comment by squidfunk** [comment]
Thanks for suggesting. I think we talked about this in the Gitter channel some time ago. I haven't seen this anywhere, and wonder if it's useful to many users. Let's see if this issue gets some feedback and then decide whether to implement 😊

**Comment by ktomk** [comment]
If I may chime in, when considering such, I've played around to limit the height of the code blocks (plain css, just having a scroll bar) so perhaps this is also useful to keep in mind for the expansion.

Right now my expand is to copy to clipboard and then use the editor of choice :) - Happy new year btw.

**Comment by squidfunk** [comment]
> Right now my expand is to copy to clipboard and then use the editor of choice :)

This is also my thought. I don't know if it makes much sense to add this feature, because more often than usual you're reading documentation alongside the editor of your choice and copying is dead-simple. Furthermore, this feature will only make sense for anything above the tablet portrait breakpoint (`768px`), because below that the sidebars are hidden so you wouldn't gain anything in width.

**Comment by squidfunk** [comment]
__Update__: I'm currently not considering this feature ready for implementation. Here's why:

1. As the height of code blocks is dynamic and will always show all lines, it will only apply to code blocks that are too wide and which are rendered with a scrollbar. Below `960px` viewport width (mobile + tablets in portrait mode), code blocks are already rendered full width. Thus, it will only impact tablet in landscape and desktop breakpoints. It even doesn't make sense to show the expand button below `960px`, as it will render the exact same code block.

2. I understand that desktop users make up a huge fraction (at least it's 90% for the official Material for MkDocs documentation). However, if a user reads documentation on a desktop computer, it's not a big problem to copy & paste a code block that runs out of width into the editor of choice.

3. Code blocks show 80 characters on the desktop breakpoint, which is more or less the de-facto standard for printing code. If wider code blocks are desired, the [content area width](https://squidfunk.github.io/mkdocs-material/setup/setting-up-navigation/#content-area-width) can be adjusted with a few lines of CSS, or the table of contents can be [integrated into the navigation](https://squidfunk.github.io/mkdocs-material/setup/setting-up-navigation/#navigation-integration). Furthermore, lines can be wrapped with some CSS, if they are too long.

    <img width="702" alt="Bildschirmfoto 2021-01-19 um 18 15 12" src="https://user-images.githubusercontent.com/932156/105069702-97905900-5a82-11eb-85b7-39c987ca849e.png">

That said, there are multiple ways to workaround the problem. For this reason, I'm closing this issue.

**Comment by shu307** [comment]
It's been a long time but maybe [**this**](https://github.com/squidfunk/mkdocs-material/discussions/4370) is what you need.

**Comment by hellt** [comment]
If anyone has skills to deliver this as a plugin a monetary compensation, please reach me at `roman at dodin dot dev`

**Comment by hellt** [comment]
Here is a good implementation example on dev.to blogging platform


https://github.com/squidfunk/mkdocs-material/assets/5679861/42c64a29-8e66-47fd-9fb2-190a48a2ba42

**Comment by hellt** [comment]
Riding the vibe waves we added this yesterday. Most certainly, it is not an optimal code, but it achieves the task and does what we need.

https://github.com/nokia-eda/docs/commit/f740e38c2960918a329d5ecbd9d21a715420482d

We added this for code blocks and tables - the two specimens that cry for being displayed on a wider viewbox than the original reading container.

https://github.com/user-attachments/assets/93141f49-d285-4eac-8e35-5714519e4b2b
